python/CMakeLists.txt

```diff
@@ -17,34 +17,30 @@
 # under the License.
 #
 
-# NB For python the SWIG module name must have the same name as the input .i file for CMake to generate the
-# correct dependencies
-
-set(CMAKE_SWIG_FLAGS "-threads" "-DUINTPTR_SIZE=${CMAKE_SIZEOF_VOID_P}")
-
 include_directories (${PN_C_INCLUDE_DIR} ${Python_INCLUDE_DIRS})
 
-set_source_files_properties(cproton.i PROPERTIES CPLUSPLUS NO)
-
-# Suppress warnings in swig generated code.
-set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
-
-# Set Compiler extra flags for Solaris when using SunStudio
-if (CMAKE_C_COMPILER_ID STREQUAL "SunPro")
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DSWIGEXPORT=__global")
-endif ()
-
+add_custom_command(
+        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/_proton_core.c
+        COMMAND ${Python_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/proton_cffi_builder.py" "${CMAKE_CURRENT_BINARY_DIR}/_proton_core.c"
+        DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/proton_cffi_builder.py)
 
-list(APPEND SWIG_MODULE_cproton_EXTRA_DEPS
-    ${CMAKE_SOURCE_DIR}/c/include/proton/cproton.i
-    ${PROTON_HEADERS}
-)
+if (WIN32)
+  set(python_suffix .pyd)
+else ()
+  set(python_suffix .so)
+endif()
 
-swig_add_library(cproton LANGUAGE python SOURCES cproton.i)
-swig_link_libraries(cproton ${BINDING_DEPS} ${Python_LIBRARIES} -lm)
-set_target_properties(${SWIG_MODULE_cproton_REAL_NAME}
+add_library(_proton_core SHARED ${CMAKE_CURRENT_BINARY_DIR}/_proton_core.c)
```

Comment:
Appveyor:

```
CMake Error at C:/Tools/vcpkg/scripts/buildsystems/vcpkg.cmake:702 (_add_library):
  _add_library cannot create target "_proton_core" because another target
  with the same name already exists.  The existing target is a shared library
  created in source directory "C:/projects/qpid-proton/python3".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  python/CMakeLists.txt:33 (add_library)
CMake Error at python/CMakeLists.txt:35 (target_link_libraries):
  Attempt to add link library "qpid-proton-core" to target "_proton_core"
  which is not built in this directory.
  This is allowed only when policy CMP0079 is set to NEW.
```